6. Binary Tree/collect_target/CollectTargetMerlin.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * CollectTargetMerlin.java
+ * Check if any root to leaf path has a given sum
+ *
+ * Description-
+ * For any node, number of coins from root to that node=total coins from root to its parent+coins in that node
+ * Hence we calculate total path sums for all leaf nodes
+ * Our ans will be true if a path sum=desired sum and no otherwise
+ *
+ * Time Complexity-O(n), Space Complexity-O(n)
+ *
+ * @author merlin[https://github.com/m-e-r-l-i-n]
+ */
+
+import java.io.*;
+import java.util.*;
+
+class CollectTargetMerlin
+{
+    public static void main(String args[])throws Exception
+    {
+        BufferedReader bu=new BufferedReader(new InputStreamReader(System.in));
+        StringBuilder sb=new StringBuilder();
+        int n=Integer.parseInt(bu.readLine());
+        ArrayList<Integer> g[]=new ArrayList[n];
+        int i,sum[]=new int[n],par[]=new int[n];
```

Comment:
What is par here?

Reply:
parent of any node

Reply:
array to store parent of any node(since i'm using an adjacency list to store node indices)